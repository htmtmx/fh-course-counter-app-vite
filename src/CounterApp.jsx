import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ title = 'No se ha proporcionado un tÃ­tulo', value = 0 }) => {

  const [counter, setCounter] = useState(value);

  const handleClick = (accionType) => {
    switch (accionType) {
      case 1:
        setCounter((c) => c + 1);
      break;
      case 2:
        setCounter(value);
      break;
      case 3:
        setCounter((c) => c - 1);
      break;
      default:
        break;
    }
  }

  return (
    <>
      <h1>{ title }</h1>
      <h2 data-testid='valueCounter'>{ counter }</h2>
      <button
        onClick={()=>handleClick(1)}
      >
        +1
      </button>
      <button aria-label='btn-reset'
        onClick={()=>handleClick(2)}
      >
        Reset
      </button>
      <button 
        onClick={()=>handleClick(3)}
      >
        -1
      </button>
    </>
  )
}

CounterApp.propTypes = {
  title: PropTypes.string,
  value: PropTypes.number.isRequired,
}