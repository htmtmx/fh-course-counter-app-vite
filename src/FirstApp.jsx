export const FirstApp = () => {
  return (
    <h1>Cesar</h1>
  )
}
