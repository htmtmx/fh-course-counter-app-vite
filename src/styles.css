html, body{
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button{
  background-color: #363636;
  border-radius: 5px;
  border: none;
  color: white;
  cursor: pointer;
  font-size: 1.3rem;
  margin: 1.5rem 0.5rem;
  padding: 10px 20px;
}