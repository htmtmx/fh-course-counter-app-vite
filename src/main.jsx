import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import CssBaseline from '@mui/material/CssBaseline';
import './styles.css'

const darkTheme = createTheme({
  palette: {
    mode: 'dark',
  },
});

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <ThemeProvider theme={darkTheme}>
      <CssBaseline/>
      <FirstApp title='Usando PropTypes' subtitle='Lorem ipsum dolor sit amet consectetur, adipisicing elit. Itaque omnis eius neque. Veritatis, vitae voluptas! Magni ad, soluta dolor optio earum accusamus odio impedit nam, quidem similique nobis, obcaecati id!' />
      
    </ThemeProvider>
  </React.StrictMode>
);