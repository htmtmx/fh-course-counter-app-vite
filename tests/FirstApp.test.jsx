import { render } from "@testing-library/react";
import { FirstApp } from '../src/FirstApp';

// eslint-disable-next-line no-undef
describe('Pruebas en FirstApp', () => {
  // eslint-disable-next-line no-undef
  test('Debe hacer match con el snapshot', () => {
    
    const title = 'Presentación:';
    const name = 'Hola soy Cesar'
    const { container } = render(<FirstApp name={name} title={title} />);
    // eslint-disable-next-line no-undef
    expect(container).toMatchSnapshot();
  });
  // eslint-disable-next-line no-undef
  test('Debe de mostrar el titulo en un h1', () => { 
    const title = 'Presentación:';
    const name = 'Hola soy Cesar'
    const { container, getByText } = render(<FirstApp name={name} title={title} />);
    // eslint-disable-next-line no-undef
    expect(getByText(title)).toBeTruthy();
    
    const h1 = container.querySelector('h1');
    // eslint-disable-next-line no-undef
    expect(h1.innerHTML).toContain(title);
    // expect(h1)
   })
});