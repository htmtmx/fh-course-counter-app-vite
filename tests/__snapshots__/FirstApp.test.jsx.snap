// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Presentación:
  </h1>
  <h1>
    Hola soy Cesar
  </h1>
  <p>
    Sin subtítulo
  </p>
</div>
`;
